File: frontend/src/types/category.ts
export interface Category {
  id: number
  name: string
  slug: string
  description: string | null
  is_active: boolean
}

export interface CategoryCreate {
  name: string
  description?: string | null
}

export interface CategoryUpdate {
  name: string
  description?: string | null
  is_active: boolean
}

File: backend/app/schemas/category.py
from pydantic import BaseModel


class CategoryCreate(BaseModel):
    name: str
    description: str | None = None


class CategoryUpdate(BaseModel):
    name: str
    description: str | None = None
    is_active: bool


class CategorySummary(BaseModel):
    id: int
    name: str
    slug: str

    model_config = {
        "from_attributes": True
    }


class CategoryResponse(BaseModel):
    id: int
    name: str
    slug: str
    description: str | None
    is_active: bool

    model_config = {
        "from_attributes": True
    }